var/spack/repos/builtin/packages/bat/package.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Bat(Package):
+    """A cat(1) clone with wings."""
+
+    homepage = "https://github.com/sharkdp/bat"
+    url      = "https://github.com/sharkdp/bat/archive/v0.10.0.tar.gz"
+
+    version('0.12.1', sha256='1dd184ddc9e5228ba94d19afc0b8b440bfc1819fef8133fe331e2c0ec9e3f8e2')
+
+    depends_on('rust')
+
+    def install(self, spec, prefix):
+        cargo = which('cargo')
+        cargo('install', '--root', prefix, '--path', '.')
+
+    # needed for onig_sys
+    def setup_build_environment(self, env):
+        env.append_flags('LLVM_CONFIG_PATH',
+                         join_path(self.spec['llvm'].prefix.libexec.llvm,
+                                   'llvm-config'))
+        env.append_flags('LIBCLANG_PATH', self.spec['llvm'].prefix.lib)
```

Comment:
Is this really specific to rust? It looks like it should be somehow in the llvm package?

Reply:
I don't have a good answer for that.  I think that what's going on is that onig_sys is taking advantage of its knowledge that Rust is implemented as a front end to LLVM and using some of it's tools.  I'm not sure if that's a bit of a privacy/boundary violation or....  

I took a look at the LLVM package and it's not setting/exporting either of those variables into dependent environments, presumably the actual LLVM tool chain knows how to find them when it needs them.

In other words, I *think* that this does belong here, specifically in things that use onig_sys, rather than as a general Rust or LLVM thing.

Reply:
Could you put a TL;DR of this explanation as a comment inside `setup_build_environment`? So that people know, why this is here and why it's specific to this?

If we have more users of onig_sys, we maybe could have some sort of package (possibly using `BundlePackage`), so that others can `depends_on('onig_sys')` and that package then doing some setup dependent environment?

Reply:
I'll happily extend the comment that's already there above `setup_build_environment`.

There's currently no idea of spack packages for things that cargo is dragging in when it builds Rust applications.  There was some discussion [over in this sibling PR](https://github.com/spack/spack/pull/15343#discussion_r387992194).

I'm also only mostly certain that this is something funny about onig_sys, it *might* be a bug in our Rust package or ....

Reply:
@ChristianTackeGSI -- I've updated it, does that make sense?

Reply:
Looks good!